src/c++/perf_analyzer/request_rate_manager.cc

```diff
@@ -247,6 +247,23 @@ RequestRateManager::Infer(
   do {
     // Should wait till main thread signals execution start
     if (!execute_) {
+      if (on_sequence_model_) {
+        // Finish off all the ongoing sequences for graceful exit
+        for (size_t i = thread_config->id_; i < sequence_stat_.size();
+             i += thread_config->stride_) {
+          std::lock_guard<std::mutex> guard(sequence_stat_[i]->mtx_);
+          sequence_stat_[i]->paused_ = true;
```

Comment:
I don't love this, but it was the only way I could come up with to avoid a race condition. Each thread can issue to ANY sequence, but here each thread is responsible for finishing specific sequences.  Without this, one thread could finish a thread, only to have another thread not reach this code yet and start a new sequence.

Note that ConcurrencyManager doesn't use `paused_` because the no sequences are shared between threads.

Reply:
Just to make sure I'm understanding this correctly, now `Infer` will be called with this same rate as before. However, the threads will skip sending inference requests since `paused_` is set to true?

Reply:
This is only to handle the corner case where one thread has been paused and has "finished" its outstanding sequences, while another thread has not yet been paused and may still be issuing a new sequence request. In that case, the not-yet-paused thread will skip issuing the new request.

After all sequences have been paused, the new request rate will be established and everything will be unpaused.

Reply:
I see. Resolving.